### PR TITLE
@mzikherman => Artwork Brick Middleware

### PR DESCRIPTION
**Pull request by damassi**
Went down a couple different routes but this turned out to be the simplest. This is a first step for the global artwork brick; subsequent PR's will start swapping out brick code with this implementation. Whats going on: 

1. Middleware is mounted and for every request attach a `renderArtworkBrick` function to `locals`
1. From template code call that function with `props`
1. Inside `renderArtworkBrick`, import Reaction's artwork brick and passes props to it, and then run `ReactDOM.renderToString` to output a mountable interface 

#### Todo in separate PRs:
- Ensure that Reaction artwork brick can be used without Relay. Right now harmless warnings are being thrown in the console due to a hack component parent wrapper I wrote a while back to get around the Relay network layer requirement (see https://artsy.slack.com/files/U3UNN0FQT/FA7QQJ12P/screen_shot_2018-04-16_at_7.32.52_pm.png)
- Ensure Reaction artwork brick utils like favoriting can be used without relay
- Swap out bricks where encountered in code

### CODE REVIEW

File: src/desktop/components/react/reaction_artwork_brick/index.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    },
+    href: '/artwork/mikael-olson-some-kind-of-dinosaur',
+    ...props,
+  }
+
+  try {
+    const out = renderToString(
+      <RelayStubProvider>
+        <Artwork artwork={artworkProps} />
+      </RelayStubProvider>
```
Comment (damassi) [inline src/desktop/components/react/reaction_artwork_brick/index.js]: Hack to get around Relay network requirements

File: src/lib/template_helpers.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    const foundFlag = Boolean(sd[flag])
+
+    if (foundFlag) {
+      classes.push(flagMap[flag])
+    }
+  })
+
+  const bodyClass = classes.join(' ')
+  return bodyClass
+}
```
Comment (damassi) [inline src/lib/template_helpers.js]: Convert from coffee to JS

**Comment by damassi** [comment]
(Merging from offline convo with @mzikherman 👍)

File: src/desktop/components/react/reaction_artwork_brick/index.js (hunk 1 above)
Comment (mzikherman) [reply src/desktop/components/react/reaction_artwork_brick/index.js]: 👍

File: src/desktop/lib/middleware/renderArtworkBrick.js [hunk 3]
```diff
@@ -0,0 +1,8 @@
+const {
+  renderArtworkBrick,
+} = require('../../components/react/reaction_artwork_brick')
```
Comment (mzikherman) [inline src/desktop/lib/middleware/renderArtworkBrick.js]: We might be able to use absolute paths here?

**Comment by mzikherman** [comment]
This looks like a fantastic approach/start to me! 🍾

**Pull request merged**